shared/pingpong/pingpong.go

```diff
@@ -696,8 +739,7 @@ func (pps *WorkerState) sendFromTo(
 		*ap = p
 		assetsByCreator[c] = append(assetsByCreator[c], ap)
 	}
-	for i := 0; i < len(fromList); i = (i + 1) % len(fromList) {
-		from := fromList[i]
+	for i, from := range fromList {
```

Comment:
I thought infinite loop was by design. Is it not?

Reply:
If it was then it should have been `for {` and explicitly an infinite loop. This really looks like an accidental infinite loop to me. OR, it _was_ by design, but I found a case where that design doesn't work. In the outer loop outside of this function there are checks to run for a total number of seconds then exit pingpong. When I tried to use that it wouldn't work because this infinite loop wouldn't break until 1/5 of the accounts ran out of money. So, just do one pass through the from list, and let the outer loop do its thing.